test/powershell/Modules/Microsoft.PowerShell.Core/PSSessionConfiguration.Tests.ps1

```diff
@@ -0,0 +1,854 @@
+                    $Result = [PSObject]@{Session = Get-PSSessionConfiguration -Name $TestSessionConfigName; Culture = (Get-Item WSMan:\localhost\Plugin\$endpointName\lang -ea SilentlyContinue).value}
+
+                    $Result.Session.Name | Should be $TestSessionConfigName
+                    $Result.Session.SessionType | Should be "Default"
+                    $Result.Session.PSVersion | Should be 6.0
+                    $Result.Session.Enabled | Should be $true
+                    $Result.Session.lang | Should be $Result.Culture
+                    $Result.Session.pssessionthreadoptions | Should be $pssessionthreadoptions
+                    $Result.Session.psmaximumreceivedobjectsizemb | Should be $psmaximumreceivedobjectsizemb
+                    $Result.Session.psmaximumreceiveddatasizepercommandmb | Should be $psmaximumreceiveddatasizepercommandmb
+                    $Result.Session.UseSharedProcess | Should be $UseSharedProcess
+                }
+                
+                It "Validate Register-PSSessionConfiguration -startupscript parameter" {
+
+                    $null = Register-PSSessionConfiguration -Name $TestSessionConfigName -path $LocalConfigFilePath -StartupScript $LocalStartupScriptPath -Force
+
+                    ValidateRemoteEndpoint -TestSessionConfigName $TestSessionConfigName -ScriptToExecute "return `$script:testvariable" -ExpectedOutput "testValue" -ExpectedError $null
+                }
+                
+                
+                It "Validate Register-PSSessionConfiguration -AccessMode parameter" {
+
+                    $null = Register-PSSessionConfiguration -Name $TestSessionConfigName -path $LocalConfigFilePath -AccessMode Disabled -Force
+
+                    ValidateRemoteEndpoint -TestSessionConfigName $TestSessionConfigName -ScriptToExecute $null -ExpectedOutput $null -ExpectedError "RemoteConnectionDisallowed,PSSessionOpenFailed"
+                }
+                
+                
+                It "Validate Register-PSSessionConfiguration -ModulesToImport parameter" {
+
+                    $null = Register-PSSessionConfiguration -Name $TestSessionConfigName -ModulesToImport $LocalTestModulePath -Force
+                    
+                    ValidateRemoteEndpoint -TestSessionConfigName $TestSessionConfigName -ScriptToExecute "return IsTestModuleImported" -ExpectedOutput $true -ExpectedError $null
+                }
+
+                It "Validate Register-PSSessionConfiguration with ApplicationBase, AssemblyName and ConfigurationTypeName parameter" {
+
+                    $TestAssemblyPath = join-path $LocalTestDir $LocalTestAssemblyName
+                    add-type -path $script:SourceFile -OutputAssembly $TestAssemblyPath
+                    $null = Register-PSSessionConfiguration -Name $TestSessionConfigName -ApplicationBase $LocalTestDir -AssemblyName $LocalTestAssemblyName -ConfigurationTypeName "PowershellTestConfigNamespace.PowershellTestConfig" -force
```

Comment:
Why cannot we use the same dll generated from BeforeAll?

Plus, `$LocalTestDir` is in `$TestDrive` and Pester will try removing the folder, which will result in errors once the assembly at `$TestAssemblyPath` is loaded.